pkg/controller/operators/validatingroundtripper/validating_round_tripper.go

```diff
@@ -13,23 +18,74 @@ import (
 
 type validatingRoundTripper struct {
 	delegate http.RoundTripper
+	codecs   serializer.CodecFactory
+}
+
+func (rt *validatingRoundTripper) decodeYAMLOrJSON(data []byte) (*unstructured.Unstructured, error) {
+	bodyBuffer := bytes.NewBuffer(data)
+	dec := yaml.NewYAMLOrJSONDecoder(bodyBuffer, 10)
+	unstructuredObject := &unstructured.Unstructured{}
+	if err := dec.Decode(unstructuredObject); err != nil {
+		return nil, fmt.Errorf("error decoding yaml/json object to an unstructured object: %w: %+v", err, string(data))
+	}
```

Comment:
Same question for YAML/JSON/Protobuf: is it always a guarantee that the request body contains a single object? Or do we need to handle multiple objects and continuously call `Decode` until we see `io.EOF`?

Reply:
The old code seems to expect a single object. I guess because it's a POST? does that assumption stand?

Reply:
If old code expects a single object, we're probably safe to continue with that assumption.